crypto/interface.go

```diff
@@ -59,6 +61,9 @@ type KeyStore interface {
 
 	// Delete removes the private key with the given KID from the KeyStore.
 	Delete(ctx context.Context, kid string) error
+
+	// Link links the key in the keystore to a kid
+	Link(ctx context.Context, kid string, keyName string, version string) error
```

Comment:
this will take some work since multiple modules are using it. The problem is the existence of the `SavePrivateKey` which is really only needed for some tests that have a predefined test key. I'll create an issue for this. #3292